### PR TITLE
Add method to validate keyring password

### PR DESCRIPTION
**Pull request by gantunesr**
**Add method to validate keyring password**

This PR extends the KeyringController functionality by adding a method to validate the keyring password.

**Description**

_Itemize the changes you have made into the categories below_

- BREAKING:

  - NA

- FIXED:

  - NA

- CHANGED:

  - NA

- REMOVED:

  - NA

- ADDED:

  - `KeyringController.validatePassword`
      - Method to validate the keyring password against an input

- DEPRECATED:

  - NA

- SECURITY:

  - NA

**Checklist**

- [x] Tests are included if applicable
- [x] Any added code is fully documented

**Issue**

Related to https://github.com/MetaMask/metamask-mobile/issues/4781

### CODE REVIEW

File: src/keyring/KeyringController.ts [hunk 1]
```diff
@@ -282,6 +282,16 @@ export class KeyringController extends BaseController<
     }
   }
 
+  /**
+   * Method to validate a password against the password from the keyring.
+   *
+   * @param password - Password of the keyring.
+   * @returns Boolean indicating if input password is valid
+   */
+  validatePassword(password: string): boolean {
```
Comment (Cal-L) [inline src/keyring/KeyringController.ts:291]: Just one small nit pick
There are a couple of those places in the app where has this exact logic. Should we replace those places with this?

**Review by Cal-L — APPROVED** [review]
Just left a small comment. Other than that, LGTM

File: src/keyring/KeyringController.ts (hunk 1 above)
Comment (gantunesr) [reply src/keyring/KeyringController.ts:291]: Yep [2e83073](https://github.com/MetaMask/controllers/pull/884/commits/2e83073335793f4ccfa3c7a898cd5cd8db32053f)

**Review by Cal-L — APPROVED** [review]
LGTM

**Review by Gudahtt — APPROVED** [review]
This seems OK for now, but I am not sure that that password property was meant to be public. In the extension we have been using the `verifyPassword` method for this instead. It's slower, but it has the advantage of working even when the vault is locked.

**Comment by gantunesr** [comment]
Thanks for pointing that out @Gudahtt, I wasn't aware of it. I'll open another PR to improve this solution

**Pull request merged**